pisek/jobs/parts/task_job.py

```diff
@@ -77,18 +78,24 @@ def globs_to_files(
         return [TaskPath.from_abspath(directory.path, file) for file in files]
 
     @staticmethod
-    def _short_text(text: str, max_lines: int = 15, max_chars: int = 100) -> str:
-        short_text = []
-        for i, line in enumerate(text.split("\n", max_lines)):
-            if i < max_lines:
-                if len(line) > max_chars:
-                    line = f"{line[:max_chars-3]}..."
-                short_text.append(line)
-            else:
-                short_text[-1] = "[...]\n"
-                break
-
-        return "\n".join(short_text)
+    def _short_text(
+        text: str,
+        style: Literal["h", "t", "ht"] = "h",
+        max_lines: int = 10,
+        max_chars: int = 100,
+    ) -> str:
+        s_text = []
+        for line in text.split("\n"):
+            line = line.strip()
+            if len(line) > max_chars:
+                line = line[: max_chars - 1] + "…"
+            s_text.append(line)
+        if len(s_text) < max_lines:
+            return text
+
+        tail = max(ceil(("t" in style) * max_lines / len(style)) - 1, 0)
+        head = max_lines - tail - 1
+        return "\n".join(s_text[:head] + ["[...]"] + (s_text[-tail:] if tail else []))
```

Comment:
We should probably also use the unicode ellipsis character here?

```suggestion
        return "\n".join(s_text[:head] + ["[…]"] + (s_text[-tail:] if tail else []))
```